### ISSUE TITLE
BUGS: noUnusedLocals doesn't check arguments of function

### ISSUE THREAD

**Issue by microshine** [Microsoft/TypeScript#11429]
**TypeScript Version:**  2.0.3

I'm using flag `--noUnusedLocals` for my project.

**Code**

```ts
protected static exportJwkPublicKey(session: Session, key: CryptoKey, callback: (err: Error, data: JsonWebKey) => void) {
    return new Promise((resolve, reject) => {
        let pkey: ITemplate = (<CryptoKey>key).key.getAttribute({
            publicExponent: null,
            modulus: null
        });
        let alg = this.jwkAlgName(key.algorithm);
        let jwk: JsonWebKey = {
            kty: "RSA",
            alg: alg,
            ext: true,
            key_ops: key.usages,
            e: Base64Url.encode(pkey.publicExponent as Uint8Array),
            n: Base64Url.encode(pkey.modulus as Uint8Array)
        };
        return jwk;
    });
}
```

**Expected behavior:**
`callback` is unused parameter in `exportJwkPublicKey`  function. But typescript doesn't show error for it.

**Actual behavior:** 
![image](https://cloud.githubusercontent.com/assets/7333669/19167305/b1881574-8c13-11e6-99bc-d1bf16dc76d4.png)

**Comment by bman61** [comment]
Have you tried using the --noUnusedParameters option?

**Comment by mhegazy** [comment]
`--noUnusedParameters` is what checks for unused parameters. `noUnusedLocals` checks for unused local declarations, e.g. variables.

**Comment by microshine** [comment]
Thank you. It's working. I thought that only one flag should check unused params. I was wrong
